editor/src/messages/tool/tool_messages/path_tool.rs

```diff
@@ -285,11 +343,34 @@ impl Fsm for PathToolFsmState {
 				}
 
 				// Mouse up
-				(_, PathToolMessage::DragStop { shift_mirror_distance }) => {
+				(state, PathToolMessage::DragStop { shift_mirror_distance } | PathToolMessage::Enter { shift_mirror_distance }) => {
+					let shift_pressed = input.keyboard.get(shift_mirror_distance as usize);
+
+					if state == PathToolFsmState::DrawingBox {
```

Comment:
Please move this as a new match arm above here, i.e. `(PathToolFsmState::DrawingBox, PathToolMessage::DragStop { shift_mirror_distance } | PathToolMessage::Enter { shift_mirror_distance }` to keep all match arms a reasonable size.